### PR TITLE
Feature/bootstrap should promote

### PR DESCRIPTION
**Pull request by iStrike7**

### CODE REVIEW

File: pallets/bootstrap/src/lib.rs [hunk 1]
```diff
@@ -482,6 +515,10 @@ pub mod pallet {
 		TokenIdDoesNotExists,
 		/// Token activations failed
 		TokensActivationFailed,
+		/// Bootstrap not schduled
```
Comment (mateuszaaa) [inline pallets/bootstrap/src/lib.rs:518]: typo

File: pallets/xyk/src/lib.rs [hunk 2]
```diff
@@ -2826,4 +2826,17 @@ impl<T: Config> mp_bootstrap::RewardsApi for Pallet<T> {
 			Some(ActivateKind::AvailableBalance),
 		)
 	}
+
+	fn promote_pool(liquidity_token_id: TokenId) -> DispatchResult {
+		ensure!(
+			<T as Config>::PoolPromoteApi::get_pool_rewards(liquidity_token_id).is_none(),
+			Error::<T>::PoolAlreadyPromoted,
+		);
```
Comment (mateuszaaa) [inline pallets/xyk/src/lib.rs:2834]:
```suggestion
		ensure!(!Self::is_promoted_pool(liquidity_asset_id), Error::<T>::PoolAlreadyPromoted);

```

**Review by mateuszaaa — APPROVED** [review]
lgtm consider adding unit test for promoting & not promoting the pool when bootstrap is finished. there is configurable mock available for that already

File: pallets/xyk/src/lib.rs (hunk 2 above)
Comment (iStrike7) [reply pallets/xyk/src/lib.rs]: Changed function

File: pallets/bootstrap/src/lib.rs (hunk 1 above)
Comment (iStrike7) [reply pallets/bootstrap/src/lib.rs]: edited. ty.

**Pull request merged**